### PR TITLE
Scale and resize window according to screen dpi

### PR DESCRIPTION
**Pull request by atomic-junky**
When a screen has a hidpi, the window becomes very small, which makes it rather complicated to use at such a small size. The solution is to calculate the scale of the window according to the dpi of the screen.

## Summary by Sourcery

Scale the application window according to the screen's DPI to ensure proper sizing on high-DPI displays and enhance user experience by centering the window upon resize.

New Features:
- Implement dynamic window scaling based on screen DPI to improve usability on high-DPI displays.

Enhancements:
- Add functionality to center the window upon resizing to maintain a consistent user experience.

### CODE REVIEW

File: Scripts/Global/App.gd [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+func update_window(update_size: bool = false):
+	var screen_dpi = DisplayServer.screen_get_dpi()
+	var screen_size = DisplayServer.window_get_size()
+	# Guess the screen scale because [DisplayServer.screen_get_scale()] only work on macOS and Linux.
+	var guessed_screen_scale: float = min(round(screen_dpi/BASE_SCALE_DPI), 2.0)
+	
+	ProjectSettings.set_setting("display/window/stretch/scale", guessed_screen_scale)
+	ProjectSettings.save_custom("user://override.cfg")
```
Comment (RailKill) [inline Scripts/Global/App.gd:20]: This doesn't do anything when you export the project, it only works when you run Monologue from the Godot editor. When using `override.cfg`, it should **"be used in exported projects by placing this file in the same directory as the project binary"**, not in the user directory, see [ProjectSetttings overriding](https://docs.godotengine.org/en/stable/classes/class_projectsettings.html#projectsettings).

> To configure the stretch scale at runtime from a script, use the `get_tree().root.content_scale_factor` property (see [Window.content_scale_factor](https://docs.godotengine.org/en/stable/classes/class_window.html#class-window-property-content-scale-factor)).

**Review by RailKill — CHANGES_REQUESTED** [review]
It's a great idea but it is pretty hard to use on windowed mode in big screens (2560x1440):

ui-scaling1

Even on fullscreen, I feel that the UI scaling is too much and there is some quality loss on some of the controls. Main issue is that it makes windowed mode a poor experience on large screens.

**Comment by atomic-junky** [comment]
Okay, I'll try to find a solution.

**Comment by atomic-junky** [comment]
Does it work better now? I took over the Godot c++ code logic.

*Btw I created a discussion (in the discussion tab) I don't know if it has any interest but it does exist*

**Review by RailKill — APPROVED** [review]
Scaling works fine now, nice!

I tested it on lower resolutions, the window scales down to fit the resolution, and all controls in the side node panel are shown perfectly, everything works great. Compared to older version of Monologue, this scaling did not happen, so I can see the change clearly here. On 2560x1440 (that's the max I can go on my 27" display), the size of the UI remains the same as before, maintaining the experience. I looked at the code and presumably, since it worked for smaller resolutions on my display, it should scale up fine for bigger screens, following the if-else code logic. Great job! 👍

**Pull request merged**